tests/e2e/specs/backend/site-editing-templates.test.js

```diff
@@ -167,6 +170,8 @@ describe( 'Store Editing Templates', () => {
 			await visitTemplateAndAddCustomParagraph( 'single-product' );
 
 			await goToSiteEditor( '?postType=wp_template' );
+			// we need to wait for the selctor to show up, sometimes the loading is delayed and test becomes falky
+			await page.waitForSelector( SELECTORS.templates.templateActions );
```

Comment:
In the interest of separating business logic from testing (like I mentioned in the E2E workshop), it'd be better to move this line within `goToSiteEditor`. Basically, it's `goToSiteEditor` responsibility to make sure we are in the right place and give back control to the program only when we are (and handle errors when we aren't).

Reply:
But these wouldn't be there always. It's not a generic check. These would appear only if there were any edited templates available. Should we provide "states" along the path to distinguish what to wait for? We'd probably need another helper ie. goToSiteEditorTemplates(target: 'template' | 'parts', hasEdits: boolean) or something - maybe then it would make sense. 

Should we make it a separate issue to extract these as a separate helpers and think bit more about what utils we actually need there? These page objects would be great for this!

Reply:
Or maybe if we were extracting this to a separate hellper util we could use the `waitForRespponse` to ensure the templates data was loaded like I did it with the `saveTemplate`? Templates data is loaded async after the page was loaded first so we could improve it like so - should make it much more stable and maybe allow us to get rid of the try-catch?

Reply:
Alright, because I know this is a blocker right now for many people, let's merge it. I'd be thankful if you created a new issue so we won't forget about this. Maybe link this thread. 🙏